ports/espressif/mpconfigport.h

```diff
@@ -53,6 +53,12 @@
 #define MICROPY_NLR_SETJMP                  (1)
 #define CIRCUITPY_DEFAULT_STACK_SIZE        0x6000
 
+// Temporarily disable settable pystack
+#if defined(CONFIG_IDF_TARGET_ESP32C3) || defined(CONFIG_IDF_TARGET_ESP32S3)
+#undef CIRCUITPY_SETTABLE_PYSTACK
```

Comment:
Why do you need to undef this?

Reply:
Yes. It's already defined up above. Without the undef it errors out.

Reply:
We try not to test on defined/undefined. We use `#if SOME_FLAG` to test, and throw warnings if `SOME_FLAG` is not defined.

Two things to do:
1. in `circuitpy_mpconfig.h`, do:
```c
#ifndef CIRCUITPY_SETTABLE_PYSTACK
#define CIRCUITPY_SETTABLE_PYSTACK   (1)
```
and move it down into the conditionalized settings in that file, alphabetically.

2. Here in `mpconfigport.h`, do:
```c
#if defined(CONFIG_IDF_TARGET_ESP32C3) || defined(CONFIG_IDF_TARGET_ESP32S3)
#define CIRCUITPY_SETTABLE_PYSTACK     (0)
#endif
```
`mpconfigport.h` is included before `circuitpy_mpconfig.h`, so `circuitpy_mpconfig.h` will pick up the turn-off setting to 0 and not set it to 1.